src/components/SegmentedControl/component.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useCallback } from 'react';
+
+import { Testable, useBuildTestId } from '../../modules/test-ids';
+
+import { Container, Element, Size } from './styled';
+
+export type Props = Omit<React.AllHTMLAttributes<HTMLElement>, 'as' | 'size'> &
+  Testable & {
+    children: React.ReactNode[];
+    selectedIndex?: number;
+    size?: Size;
+  };
+
+export const Component = ({
+  children,
+  onChange,
```

Comment:
I think you need to omit `onChange` from `AllHTMLAttributes` and define properly with the types this component uses. Also, shouldn't we send back the new selectedIndex to `onChange` if that's what we require as a prop for `selectedIndex`?